renku/cli/dataset.py

```diff
@@ -247,70 +241,35 @@
     default='tabular',
     help='Choose an output format.'
 )
-@pass_local_client(clean=False, commit=False)
 @click.pass_context
-def dataset(ctx, client, revision, datadir, format):
+def dataset(ctx, revision, datadir, format):
     """Handle datasets."""
-    ctx.meta['renku.datasets.datadir'] = datadir
-
-    missing_dataset, missing_files = check_dataset_resources(client)
-    old_datasets = [ds for ds in dataset_pre_0_3(client)]
-
-    if missing_dataset or missing_files or old_datasets:
-        raise MigrationRequired('datasets')
+    if isinstance(ctx, click.Context):
+        ctx.meta['renku.datasets.datadir'] = datadir
 
     if ctx.invoked_subcommand is not None:
         return
 
-    if revision is None:
-        datasets = client.datasets.values()
-    else:
-        datasets = client.datasets_from_commit(client.repo.commit(revision))
-
-    DATASETS_FORMATS[format](client, datasets)
+    click.echo(dataset_parent(revision, datadir, format, ctx=ctx))
 
 
 @dataset.command()
 @click.argument('name')
-@pass_local_client(
-    clean=False, commit=True, commit_only=DATASET_METADATA_PATHS
-)
-def create(client, name):
+def create(name):
     """Create an empty dataset in the current repo."""
-    from renku.models.datasets import Creator
-
-    if write_dataset(client, name):
-
-        with client.with_dataset(name=name) as dataset:
-            click.echo('Creating a dataset ... ', nl=False)
-            creator = Creator.from_git(client.repo)
-            if creator not in dataset.creator:
-                dataset.creator.append(creator)
-
-        click.secho('OK', fg='green')
+    create_dataset(name, handle_duplicate_fn=write_dataset)
+    click.secho('OK', fg='green')
 
 
 @dataset.command()
-@click.argument('id')
-@pass_local_client(
-    clean=False, commit=True, commit_only=DATASET_METADATA_PATHS
-)
-def edit(client, id):
+@click.argument('dataset_id')
+def edit(dataset_id):
```

Comment:
We work with id's since names are not unique. Or did we made a change that names are unique now? Did I miss some other references of keyword `id`?

Reply:
```
def edit_dataset(client, dataset_id, transform_fn): --> dataset_id
def file_unlink(client, name, include, exclude): --> name
def dataset_remove(client, names, with_output=False): --> names
def export_dataset(client, id, provider, publish, tag, with_prompt=False): --> id
def tag_dataset(client, name, tag, description, force=False): --> name
def remove_dataset_tags(client, name, tags): --> name
def list_tags(client, name, format): --> name
```

Most of these calls pass name to either `def load_dataset(self, name=None):` or `def dataset_path(self, name):` which both call the parameter `name`

Reply:
Ah I see, there is a legacy reason for this mess. Let's open a ticket for this since this cleanup is also a great chunk of work. Streamlining this will require quite a few big changes, it's not just a name change.